### ISSUE TITLE
Workout Editor - Decimals in QWKCODE cause error

### ISSUE THREAD

**Issue by sticilface** [GoldenCheetah/GoldenCheetah#3846]
Essentially if the workout difficulty is scaled using the scroll bar the power levels end up with decimals.  If you then edit a single line on the workout using the properties QWKCODE window the whole workout fails. You then have to remove the 3-4 decimal places and the workout works again.  
 
Using 3.5, build 3991.

I have attached a video demonstrating the problem.  

https://user-images.githubusercontent.com/10061736/111188962-80b34200-85ad-11eb-9957-bc02916118e0.mov

**Comment by amtriathlon** [comment]
Decimals are not supported in QWKCode and they are truncated on file load, so if you scale a workout and want to use QWKCode a simple workaround is to save the workout and change back and forth so it is reloaded.
Probably we should round power targets after scaling to avoid this, but I am not sure if this has unintended consequences.

**Comment by sticilface** [comment]
I think I tried this but my workouts are .zwo as I use them in zwift. This causes them to totally break when saving and I can't reload them either.  I e currently gone back to using an online tool for this but would love to use GC as I love the W' indicator.

**Comment by amtriathlon** [comment]
As I commented before, the problem is QWKCODE only works with integer watts and scaling produces floating point watts. Format is irrelevant, same problem happens using erg or zwo.

Until this is fixed your options are: don't use scaling or don't change QWKCODE.

If you want to scale then save and reload the workout (changing back and forth, for example) so power values are truncated, and then you can change QWKCODE using integers.

**Comment by sticilface** [comment]
I think the problem is not limited to just scaling and changes to QWKCODE. 

My current workflow is to create a .zwo workout using zwofactory.  [link](https://zwofactory.com)
Download this and then load it into GC.  

A simple .zwo workout file prior to GC import looks like this
```
<workout_file>
    <author></author>
    <name>VO2max</name>
    <description></description>
    <sportType>bike</sportType>
    <tags>
    </tags>
    <workout>
        <SteadyState Duration="120" Power="0.43"/>
        <SteadyState Duration="120" Power="0.54"/>
        <SteadyState Duration="180" Power="0.65"/>
        <SteadyState Duration="1000" Power="0.9"/>
        <IntervalsT Repeat="6" OnDuration="180" OffDuration="180" OnPower="1.15" OffPower="0.7"/>
        <SteadyState Duration="1000" Power="0.9"/>
        <Cooldown Duration="300" PowerLow="0.61" PowerHigh="0.43"/>
    </workout>
</workout_file>
```

From this we can see that the .zwo format scales the 'FTP' from within ZWIFT or GC.  

So immediately after import into GC you end up with decimals in the QWKCODE:

<img width="248" alt="image" src="https://user-images.githubusercontent.com/10061736/113393315-655a7c00-938e-11eb-89f8-ff913d1e8166.png">

saving as from GC back to .zwo generates this:

```
<workout_file>
    <name>VO2max</name>
    <workout>
        <SteadyState Duration="120" PowerLow="0.426087" PowerHigh="0.426087" />
        <SteadyState Duration="120" PowerLow="0.53913" PowerHigh="0.53913" />
        <SteadyState Duration="180" PowerLow="0.647826" PowerHigh="0.647826" />
        <SteadyState Duration="1000" PowerLow="0.9" PowerHigh="0.9" />
        <IntervalsT Repeat="6" OnDuration="180" OffDuration="180" PowerOnLow="1.13913" PowerOnHigh="1.13913" PowerOffLow="0.7" PowerOffHigh="0.7" />
        <SteadyState Duration="1000" PowerLow="0.9" PowerHigh="0.9" />
        <Cooldown Duration="300" PowerLow="0.608696" PowerHigh="0.426087" />
    </workout>
</workout_file>
```
Loading this back into zwofactory generates the incorrect workout, and this behaviour is mirrored in zwift itself 
<img width="1039" alt="image" src="https://user-images.githubusercontent.com/10061736/113408092-9cd52280-93a6-11eb-8df4-085a2a3e7c20.png">

It looks to me like the descriptors for `<IntervalsT>` are wrong and should be `OnDuration`, `OffDuration`, `OnPower` and `OffPower` 

So issues:

- scaling produces decimal QWKCODE - not compatible
- opening .ZWO produces decimal QWKCODE - not compatible 
- saving .ZWO with intervals is incorrect.

**Comment by amtriathlon** [comment]
ZWO file import error was previously reported and it is fixed in v3.6 builds, see: https://github.com/GoldenCheetah/GoldenCheetah/issues/3675
